python/ray/autoscaler/_private/kuberay/autoscaling_config.py

```diff
@@ -333,6 +358,31 @@ def _get_num_gpus(
     return None
 
 
+def _get_num_tpus(
+    ray_start_params: Dict[str, str],
+    k8s_resource_limits: Dict[str, Any],
+    group_name: str,
+) -> Optional[int]:
+    """Get memory resource annotation from ray_start_params or k8s_resource_limits,
+    with priority for ray_start_params.
+    """
+
+    if "num-tpus" in ray_start_params:
+        return int(ray_start_params["num-tpus"])
+    else:
+        for key in k8s_resource_limits:
+            # e.g. google.com/tpu
+            if key.endswith("tpu"):
```

Comment:
Should we be checking the full string instead?

Reply:
I think like Allen mentioned in one of his comments, sometimes the resource is provided as simply `tpu` and sometimes the full `google.com/tpu` string is used. The current implementation here should support both of those options, but if we want to standardize the resource naming I can change it to just look for `tpu`.

Reply:
I think it's specifically `google.com/tpu` for the `k8s_resource_limits`

Reply:
The `tpu` vs `google.com/tpu` confusion is specific to the Ray resources. For kubernetes resources it should always be `google.com/tpu`.